R/sass.R

```diff
@@ -64,31 +64,39 @@
 #'   sass_file(tmp_file)
 #' ))
 sass <- function(input = NULL, options = sass_options(), output = NULL,
-  write_attachments = NA, cache = sass_cache_get()) {
+  write_attachments = NA, cache = sass_cache_get(), cache_key_extra = NULL) {
 
   if (!inherits(options, "sass_options")) {
     stop("Please construct the compile options using `sass_options()`.")
   }
   if (!is.null(cache) && !inherits(cache, "FileCache")) {
     stop("Please use NULL (no cache) or a FileCache object for `cache`.")
   }
-  if (!is.null(output) && !dir.exists(fs::path_dir(output))) {
-    stop("The output directory '", fs::path_dir(output), "' does not exist")
-  }
   if (is.null(output) && isTRUE(write_attachments)) {
     stop("sass(write_attachments=TRUE) cannot be used when output=NULL")
   }
 
   css <- NULL
   layer <- extract_layer(input)
 
-  if (!is.null(cache)) {
-    cache_key <- sass_hash(list(
-      input,
+  # If caching is active, compute the hash key
+  cache_key <- if (!is.null(cache)) {
+    sass_hash(list(
```

Comment:
Now that we have `output_file()` and `cache_key_extra` it seems like we could get away with not exporting `sass_hash()`

Reply:
Yeah, I think you're right.